lib/python/rose/apps/rose_ana.py

```diff
@@ -85,32 +85,53 @@ def create(self):
             status TEXT,
             comparison TEXT,
             PRIMARY KEY(app_task))
-            """)
-        # And this one stores the completion status of the task
-        conn.execute(
             """
+        self.execute_sql_retry(conn, (sql_statement,))
+        # And this one stores the completion status of the task
+        sql_statement = """
             CREATE TABLE IF NOT EXISTS tasks (
             task_name TEXT,
             completed INT,
             PRIMARY KEY(task_name))
-            """)
+            """
+        self.execute_sql_retry(conn, (sql_statement,))
         conn.commit()
 
+    def execute_sql_retry(self, conn, sql_args, retries=10, timeout=5.0):
+        """
+        Given a connection object and a tuple of the desired arguments;
+        calls sql execute and retries multiple times, to handle
+        concurrency issues
+
+        """
+        for retry in range(retries):
+            try:
+                conn.execute(*sql_args)
+                return
+            except:
+                time.sleep(timeout)
+        raise
```

Comment:
This is probably me being ignorant of a Python thing, but don't you need to specify what you're raising?